x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -766,6 +766,18 @@ public void loadExtensions(ExtensionLoader loader) {
      */
     public static final int MAX_LOW_PRIORITY_MODELS_PER_NODE = 100;
 
+    /**
+     * The time interval without any requests that has to pass, before scaling down
+     * to zero allocations.
+     */
+    public static final Setting<TimeValue> ADAPTIVE_ALLOCATIONS_SCALE_TO_ZERO_TIME = Setting.timeSetting(
+        "xpack.ml.adaptive_allocations_scale_to_zero",
```

Comment:
If this is just for writing integration tests, I don't think this is great practice. This setting is now exposed to users, right? Which could lead to people setting it to bad values, potentially hard-to-debug SDHs, etc. I also remember someone telling me the policy was to not add (many) settings.

Unfortunately, I don't see a better alternative...